Studio.kt

```diff
@@ -141,49 +143,59 @@ object Studio {
             onPreviewKeyEvent = { handleKeyEvent(it, ::confirmClose) },
             onCloseRequest = { if (error != null) exitApplicationFn() else confirmClose() },
         ) {
-            val density = LocalDensity.current.density
-            var titleBarHeight by remember { mutableStateOf(0.dp) }
             CompositionLocalProvider(LocalWindow provides window) {
-                Column(Modifier.fillMaxSize().background(Theme.studio.backgroundMedium).onGloballyPositioned {
-                    titleBarHeight = window.height.dp - toDP(it.size.height, density)
-                }) {
-                    CompositionLocalProvider(LocalTitleBarHeight provides titleBarHeight) {
-                        Toolbar.Layout()
-                        Separator.Horizontal()
-                        Frame.Row(
-                            modifier = Modifier.fillMaxWidth().weight(1f),
-                            separator = Frame.SeparatorArgs(Separator.WEIGHT),
-                            Frame.Pane(
-                                id = Browsers.javaClass.name,
-                                minSize = Browsers.MIN_WIDTH,
-                                initSize = Either.first(Browsers.DEFAULT_WIDTH)
-                            ) { Browsers.Layout(browsers, it, Browsers.Position.LEFT) },
-                            Frame.Pane(
-                                id = Pages.javaClass.name,
-                                minSize = Pages.MIN_WIDTH,
-                                initSize = Either.second(1f)
+                val windowContext = WindowContext(window)
+                CompositionLocalProvider(LocalWindowContext provides windowContext) {
```

Comment:
We have to provide a windowContext here as well as in MainWindowContent so that the dialogs are provided with a `WindowContext` upon initialisation.